anesthesia-monitor/src/actions/index.js

```diff
@@ -0,0 +1,42 @@
+import firebase, { reference } from '../utils/firebaseClient'
+
+export const fetchReadings = () => {
+  return dispatch => {
+    firebase.on('value', snapshot => {
+      dispatch({
+        type: 'FETCH_READINGS',
+        payload: snapshot.val()
+      })
+    })
+  }
+}
+
+export const addReading = (reading) => {
+  // create new record in firebase from info
+    // if successful, update the redux store
+    // if failure, show an error message
+  return dispatch => {
+    const id = reference.push(reading).path.o[1]
```

Comment:
Can you give me a little context on what `.path.o[1]` is supposed to be? From a quick glance, the thing that looks a bit off here is that you're not actually waiting for a successful firebase transaction to finish before you start dispatching an action.  An example of how you might add a new record to firebase and dispatch upon success can be found [here](https://github.com/turingschool-examples/react-redux-firebase/blob/master/src/actions/expenses.js#L41-L47).